app/helpers/application_helper/navbar.rb

```diff
@@ -1,5 +1,33 @@
 module ApplicationHelper
   module Navbar
+    def menu_to_json(position)
+        structure = []
+        Menu::Manager.menu(position) do |menu_section|
+            next unless menu_section
+            structure << item_to_hash(menu_section)
+        end
+        structure
+    end
+    
+    def item_to_hash(item)
+      {
+        :id             => item.id,
+        :name           => item.name,
+        :icon           => item.icon,
+        :placement      => item.placement,
+        :before         => item.try(:before),
+        :type           => item.type,
+        :href           => item.href,
+        :parent_id      => item.try(:parent_id),
+        :feature        => item.try(:feature),
+        :rbac_feature   => item.try(:rbac_feature),
+        :defaults       => item.try(:defaults),
+        :items          => item.items.to_a.map(&method(:item_to_hash)),
+        :visible        => item.visible?,
+        :link_params    => item.link_params
+      }
```

Comment:
You could implement this a little prettier:
```ruby
%i[
  id
  name
  icon
  placement
  before
  type
  href
  parent_id
  feature
  rbac_feature
  defaults
  link_params
].each_with_object(:visible => item.visible?, :items => item.items.to_a.map(&method(:item_to_hash))) do |obj, key|
  obj[key] = item.try(key)
end
```

Except of `visible` and `items` all the fields map nicely to the keys...

Reply:
10 points to Griffindor for the `&method` btw :1st_place_medal: :laughing: